### PR TITLE
Ensure MSAL and ADAL work with common core

### PR DESCRIPTION
**Pull request by kpanwar**
This monster PR brings core, adal and msal into a single project structure to allow easy development and release lifecycle, per  #935

### CODE REVIEW

**Comment by henrik-me** [comment]
> // <auto-generated>

is this auto generated? if yes, from what, e.g. are there files missing?

---
Refers to: adal/devApps/XFormsApp.Droid/Resources/Resource.Designer.cs:3 in a16b1e8. [](commit_id = a16b1e87cd1aaeae5f531ca1f410668e842e5a09, deletion_comment = False)

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireDeviceCodeHandler.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+using Microsoft.Identity.Core;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.ClientCreds;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Helpers;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Http;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Instance;
+using Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.OAuth2;
+
+namespace Microsoft.IdentityModel.Clients.ActiveDirectory.Internal.Flows
+{
+    class AcquireDeviceCodeHandler
```
Comment (henrik-me) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Flows/AcquireDeviceCodeHandler.cs]: internal ?

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Http/HttpClientWrapper.cs [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+                string reponseHeaderKey = header.Key;
+                string trimmedKey = reponseHeaderKey.Trim();
+                if (string.Compare(trimmedKey, OAuthHeader.CorrelationId, StringComparison.OrdinalIgnoreCase) == 0)
+                {
+                    string correlationIdHeader = headers.GetValues(trimmedKey).FirstOrDefault().Trim();
+                    Guid correlationIdInResponse;
+                    if (!Guid.TryParse(correlationIdHeader, out correlationIdInResponse))
+                    {
+                        var msg = string.Format(CultureInfo.CurrentCulture,
+                            "Returned correlation id '{0}' is not in GUID format.", correlationIdHeader);
```
Comment (henrik-me) [inline adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Http/HttpClientWrapper.cs]: >"Returned correlation id '{0}' is not in GUID format." [](start = 28, length = 54)

any reason why strings are not added to resources?

**Comment by henrik-me** [comment]
Does the official build and tests run/pass?

**Comment by henrik-me** [comment]
Nice, great stuff!

**Comment by kpanwar** [comment]
Official VSTS build requires changes that jean-marc suggested to hold because he plans on moving our builds to devex. When he does that he will also create clones of existing builds to support updated project structures, signing and bundling. In all probability, it will be me and jean-marc grabbing dinner and getting this done together one evening. My current goal is to make appveyor pass the build.

---
In reply to: [356146447](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/937#issuecomment-356146447) [](ancestors = 356146447)

**Comment by kpanwar** [comment]
> // <auto-generated>

These files are generated for UI controls in the applications. In case of ADAL, it generates the resource file because we have an embedded webview which is an UI control. Platform equivalent in Android native is a R.java file.

---
In reply to: [356138853](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/937#issuecomment-356138853) [](ancestors = 356138853)

---
Refers to: adal/devApps/XFormsApp.Droid/Resources/Resource.Designer.cs:3 in a16b1e8. [](commit_id = a16b1e87cd1aaeae5f531ca1f410668e842e5a09, deletion_comment = False)

File: adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Http/HttpClientWrapper.cs (hunk 2 above)
Comment (kpanwar) [reply adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Http/HttpClientWrapper.cs]: I add reusable strings to a resource/constants file. If they are only used once then, personally, I feel there is no specific advantage to moving that string to a constants file.

---
In reply to: [160296794](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/937#discussion_r160296794) [](ancestors = 160296794)

**Comment by kpanwar** [comment]
sure, we can chat after scrum

---
In reply to: [356146552](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/937#issuecomment-356146552) [](ancestors = 356146552)

**Comment by henrik-me** [comment]
Discussed in person. PR builds should be running and the last part of getting builds green are next step. 

---
In reply to: [356356959](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/937#issuecomment-356356959) [](ancestors = 356356959,356146447)

**Pull request merged**